curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-fetch-and-promises-by-building-an-fcc-authors-page/641daabed8d0584b1150c953.md

```diff
@@ -0,0 +1,195 @@
+---
+id: 641daabed8d0584b1150c953
+title: Step 28
+challengeType: 0
+dashedName: step-28
+---
+
+# --description--
+
+Some of the author bios are much longer than others. To give the cards a uniform look, you can extract the first 50 characters of each one and replace the rest with an ellipsis (...). Otherwise, you can show the entire bio.
+
+Within the paragraph element, replace `bio` with a ternary operator. For the condition, check if the length of `bio` is greater than 50. If it is, use the `.slice()` method to extract the first 50 characters of `bio` and add an ellipsis at the end. Otherwise, show the full `bio`.
+
+# --hints--
+
+You should create a `p` element.
+
+```js
+assert.exists(document.querySelector('p'));
+```
+
+Your `p` element should have the class `bio`
+
+```js
+assert.match(code, /<p\s*class=("|')bio\1>/);
+```
+
+You should check if the length of the `bio` text is greater than `50`. 
+
+```js
+assert.match(code, /<p\s*class=("|')bio\1>\$\{bio\.length\s+>\s+50/)
+```
+
+If the `bio` text is greater than `50` characters, you should extract the first 50 characters with `slice()` and replace the rest with `...`. Don't forget that indexes are zero-based.
+
+```js
+assert.match(code, /<p\s*class=("|')bio\1>\$\{bio\.length\s+>\s+50\s+\?\s+bio\.slice\(0\,\s+49\)\s+\+\s+("|')\.\.\.\2\s+\:/)
+```
+
+If the `bio` text is less than 50 characters, use the `bio` text directly.
+
+```js
+assert.match(code, /<p\s*class=("|')bio\1>\$\{bio\.length\s+>\s+50\s+\?\s+bio\.slice\(0\,\s+49\)\s+\+\s+("|')\.\.\.\2\s+\:\s+bio\}<\/p>/)
+```
```

Comment:
````suggestion
You should check if the length of the `bio` text is greater than `50`. 

```js
assert.match(code, /<p\s*class=("|')bio\1>\$\{\s*bio\.length\s*>\s*50/)
```

If the `bio` text is greater than `50` characters, you should extract the first 50 characters with `slice()` and replace the rest with `...`. Don't forget that indexes are zero-based.

```js
assert.match(code, /<p\s*class=("|')bio\1>\$\{\s*bio\.length\s*>\s*50\s*\?\s*bio\.slice\(\s*0\,\s*49\s*\)\s*\+\s*("|')\.\.\.\2\s*\:/)
```

If the `bio` text is less than 50 characters, use the `bio` text directly.

```js
assert.match(code, /<p\s*class=("|')bio\1>\$\{\s*bio\.length\s*>\s*50\s*\?\s*bio\.slice\(\s*0\,\s*49\s*\)\s*\+\s*("|')\.\.\.\2\s*\:\s*bio\s*\}<\/p>/)
```
````

All of the spaces are optional here.